src/utils/hooks/api/loans/use-get-account-positions.tsx

```diff
@@ -59,31 +68,74 @@ const getAccountPositionsStats = (
-      currency,
-      ...rest
-    }));
+  const collateralLendPositions = lendPositions.filter(({ isCollateral }) => isCollateral);
+
+  const collaterAmountPositions = collateralLendPositions.map(({ amount, currency, ...rest }) => ({
+    // MEMO: compute total value based on collateral threshold (not full lend amount value)
+    amount: amount.mul(assets[currency.ticker].collateralThreshold),
+    currency,
+    ...rest
+  }));
 
-  const collateralAmountUSD = getPositionsSumOfFieldsInUSD('amount', collateralLendPositions, prices);
+  const liquidationLendPositions = collateralLendPositions.map(({ amount, currency, ...rest }) => ({
+    // MEMO: compute total value based on collateral threshold (not full lend amount value)
+    amount: amount.mul(assets[currency.ticker].liquidationThreshold),
+    currency,
+    ...rest
+  }));
 
+  const collateralAmountUSD = getPositionsSumOfFieldsInUSD('amount', collaterAmountPositions, prices);
+  const liquidationAmountUSD = getPositionsSumOfFieldsInUSD('amount', liquidationLendPositions, prices);
+
+  const collateralizedAmountUSD = getPositionsSumOfFieldsInUSD('amount', collateralLendPositions, prices);
   const earnedInterestAmountUSD = getPositionsSumOfFieldsInUSD<LendPosition>('earnedInterest', lendPositions, prices);
   const earnedDeptAmountUSD = getPositionsSumOfFieldsInUSD('accumulatedDebt', borrowPositions, prices);
 
   const totalEarnedRewardsUSDValue =
     convertMonetaryAmountToValueInUSD(subsidyRewards, getTokenPrice(prices, subsidyRewards.currency.ticker)?.usd) || 0;
-  const netYieldAmountUSD = earnedInterestAmountUSD.add(totalEarnedRewardsUSDValue).sub(earnedDeptAmountUSD);
+  const netAmountUSD = earnedInterestAmountUSD.add(totalEarnedRewardsUSDValue).sub(earnedDeptAmountUSD);
+
+  const totalLendApy = lendPositions.reduce((total, position) => {
+    const { lendApy, lendReward } = assets[position.currency.ticker];
+    const rewardsApy = getSubsidyRewardApy(position.currency, lendReward, prices);
+    const positionApy = lendApy.add(rewardsApy || 0);
+    const positionUSDValue = convertMonetaryAmountToValueInUSD(
+      position.amount,
+      getTokenPrice(prices, position.currency.ticker)?.usd
+    ) as any;
+
+    return positionUSDValue ? total.add(positionApy.mul(positionUSDValue)) : total;
+  }, new Big(0));
+
+  const totalBorrowApy = borrowPositions.reduce((total, position) => {
+    const { borrowApy, borrowReward } = assets[position.currency.ticker];
+    const rewardsApy = getSubsidyRewardApy(position.currency, borrowReward, prices);
+    const positionApy = borrowApy.sub(rewardsApy || 0);
+    const positionUSDValue = convertMonetaryAmountToValueInUSD(
+      position.amount,
+      getTokenPrice(prices, position.currency.ticker)?.usd
+    );
+
+    return positionUSDValue ? total.add(positionApy.mul(positionUSDValue)) : total;
+  }, new Big(0));
+
+  const netAPY = totalLendApy.sub(totalBorrowApy).div(supplyAmountUSD);
```

Comment:
This computation is possibly incorrect, see:
![image](https://user-images.githubusercontent.com/47864599/205694191-4afabe0c-7d3f-4834-9fef-41a2f2dc73e3.png)
We need to test this properly:
- check that APYs are weighted properly according to USD normalized position amount
- test this with all positions created at the same time (ideally on isolated testnet so nothing can interfere with APYs) so there are no previous accumulated rewards

It seems to me that earned APY might not be correct, this positoin is 24h old, so should be around $0.53, but maybe the APY was changed recently, therfore need to test better:
![image](https://user-images.githubusercontent.com/47864599/205695337-684c5aa4-6640-4153-b2d4-2fe2fec7258d.png)

Reply:
We have to keep testing this outside of this PR. Its a scenario that does not affect negatively our protocol.